### ISSUE TITLE
Autodoc return type hints

### ISSUE THREAD

**Issue by znicholls** [openclimatedata/openscm#155]
At the moment it is impossible to use autodoc-typehints for return types (see https://github.com/sphinx-doc/sphinx/issues/5887). This means we have to write return types twice, once as a function annotation and once in the docstring. 

I've opened this issue as a reminder that this is a problem but perhaps it should just go in the contributing guide instead? Thoughts @swillner @lewisjared @rgieseke ?

**Comment by swillner** [comment]
also had that problem along the way but don't consider it much of a hassle to repeat these for now.

**Comment by znicholls** [comment]
I agree, I think I know how to add a PR into sphinx sometime too which will push the conversation along a bit faster too
